src/test/kotlin/com/exactpro/th2/codec/moldudp64/TestMoldUdp64Codec.kt

```diff
@@ -0,0 +1,529 @@
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Invalid amount of message lengths: 2 (expected: 1)", exception.message)
+        }
+
+        @Test
+        fun `mismatching message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U, packetMessageLengths = listOf(2U)),
+                message(byteArrayOf(1, 2, 3))
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Mismatching length of message 1: 2 (expected: 3)", exception.message)
+        }
+
+        @Test
+        fun `big actual message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U),
+                message(Random.nextBytes(65536))
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Actual length of message 1 is too big: 65536 (expected at most: 65535)", exception.message)
+        }
+    }
+
+    companion object {
+        private const val SESSION_FIELD = "Session"
+        private const val SEQUENCE_FIELD = "SequenceNumber"
+        private const val COUNT_FIELD = "MessageCount"
+        private const val LENGTHS_FIELD = "MessageLengths"
+
+        private fun raw(
+            packet: ByteArray,
+            th2SessionAlias: String = UUID.randomUUID().toString(),
+            th2Direction: Direction = FIRST,
+            th2Sequence: Long = System.currentTimeMillis(),
+            th2MetadataProperties: Map<String, String> = mapOf("uuid-${Random.nextULong(256U)}" to UUID.randomUUID().toString())
+        ): MessageGroup = MessageGroup.newBuilder().apply {
+            this += RawMessage.newBuilder().apply {
+                metadataBuilder.apply {
+                    putAllProperties(th2MetadataProperties)
+                    timestamp = Instant.now().toTimestamp()
+                    idBuilder.apply {
+                        connectionIdBuilder.sessionAlias = th2SessionAlias
+                        direction = th2Direction
+                        sequence = th2Sequence
+                    }
+                }
+
+                body = ByteString.copyFrom(packet)
+            }
+        }.build()
+
+        private fun header(
+            packetSession: String = System.nanoTime().toString(),
+            packetSequenceNumber: ULong = System.nanoTime().toULong(),
+            packetMessageCount: UShort = 0U,
+            packetMessageLengths: List<UShort> = listOf(),
+            th2SessionAlias: String = UUID.randomUUID().toString(),
+            th2Direction: Direction = FIRST,
+            th2Sequence: Long = Random.nextLong(),
+            th2MetadataProperties: Map<String, String> = mapOf("uuid-${Random.nextULong(256U)}" to UUID.randomUUID().toString())
+        ): Message.Builder = Message.newBuilder().apply {
```

Comment:
You can use `Message.Builder.setMetadata` extension to set most of the required metadata

Reply:
I know, but it won't make it much better. It'll be:
```kotlin
setMetadata(
    messageType = HEADER_MESSAGE_TYPE,
    sessionAlias = th2SessionAlias,
    direction = th2Direction,
    sequence = th2Sequence
)
```
vs just
```kotlin
messageType = HEADER_MESSAGE_TYPE
sessionAlias = th2SessionAlias
direction = th2Direction
sequence = th2Sequence
```
The only benefit from using it in this case is that it would automatically set timestamp to current time